core/jobs/batch_jobs/audit_exploration_jobs.py

```diff
@@ -0,0 +1,1479 @@
+        """Extracts the RTE components from an HTML string.
+
+        Args:
+            html_string: str. An HTML string.
+
+        Returns:
+            list(dict). A list of dictionaries, each representing an
+            RTE component.
+            Each dict in the list contains:
+            - id: str. The name of the component, i.e.
+            'oppia-noninteractive-link'.
+            - customization_args: dict. Customization arg specs for
+            the component.
+        """
+        components = []
+        soup = bs4.BeautifulSoup(html_string, 'html.parser')
+        oppia_custom_tag_attrs = (
+            rte_component_registry.Registry.get_tag_list_with_attrs())  # type: ignore[no-untyped-call]
+        for tag_name, tag_attrs in oppia_custom_tag_attrs.items():
+            component_tags = soup.find_all(name=tag_name)
+            for component_tag in component_tags:
+                customization_args = {}
+                for attr in tag_attrs:
+                    try:
+                        # Unescape special HTML characters such as '&quot;'.
+                        attr_val = html.unescape(component_tag[attr])
+                        customization_args[attr] = json.loads(attr_val)
+                    except Exception:
+                        continue
+
+                component = {
+                    'id': tag_name,
+                    'customization_args': customization_args
+                }
+                components.append(component)
+        return components
+
+    @staticmethod
+    def filter_invalid_rte_image_tag(
+        states_dict: Dict[str, state_domain.State]
+    ) -> List[Dict[str, List[Dict[str, List[str]]]]]:
+        """Returns the errored state values, validates
+            - alt should have at least 5 characters
+            - Image should have an SVG extension
+
+        Args:
+            states_dict: dict[str, State]. The dictionary containing
+                state name as key and State object as value.
+
+        Returns:
+            states_with_values: list[dict]. The list of dictionaries
+            containing the errored values.
+        """
+        states_with_values = []
+
+        for state_name, state in states_dict.items():
+            rte_image_errors = []
+            rte_components = ExpStateAuditChecksJob._get_rte_components(
+                state.content.html)
+            for idx, rte_component in enumerate(rte_components):
+                if rte_component['id'] == 'oppia-noninteractive-image':
+                    try:
+                        alt_with_value = rte_component['customization_args'][
+                            'alt-with-value']
+                    except Exception:
+                        alt_with_value = 'Not found'
+
+                    # Validates if alt value is greater than 5.
+                    if (
+                        alt_with_value != 'Not found' and
+                        len(alt_with_value) < 5
+                    ):
```

Comment:
Isn't the Exception case here also an error that we should log? Or did your previous PR already log these cases?

Reply:
I have covered that in PR #15748.
Thanks